src/app.js

```diff
@@ -11,36 +37,71 @@ function giveRegionColor(region) {
         return "Asia";
     } else if(region == "Oceania"){
         return "Oceania";
-        }else {
+    }else {
         return "Other";
     }
+}
+
+//-------------------------Part 2-----------------------//
+
+const form = document.getElementById("country-form");
+form.addEventListener("submit", getInputFromForm);
 
+function getInputFromForm(e) {
+    e.preventDefault();
+    let input = document.getElementById("country-name-input").value;
+    searchCountry(input);
 }
-async function giveCountry() {
+
+async function searchCountry (input) {
     try{
         const result = await
-            axios.get('https://restcountries.com/v2/all');
-            const sorted = result.data.sort((a, b) => a.population - b.population);
-        for (let i = 0; i < sorted.length ; i++) {
-            const countryDiv = document.createElement('div')
-            countryDiv.id = "country-div";
-            const countryName = document.createElement('p');
-            countryName.className = giveRegionColor(sorted[i].region);
-            const countryFlag = document.createElement('img');
-            const countryPopulation = document.createElement('p');
-            const countryPopulationNumber = sorted[i].population;
-            countryName.textContent = sorted[i].name;
-            countryFlag.src = sorted[i].flag;
-            countryPopulation.textContent = "Has a population of " + countryPopulationNumber + " people";
-            const countryList = document.getElementById("country-name");
-            countryDiv.appendChild(countryName);
-            countryDiv.appendChild(countryFlag);
-            countryDiv.appendChild(countryPopulation);
-            countryList.appendChild(countryDiv);
-        }
+            axios.get('https://restcountries.com/v2/name/' + input + "?fields=flag,name,subregion,capital,currencies,population,languages");
+        const countryInfo = result.data;
+        countryCurrency(countryInfo);
     } catch (e){
+        const searchResult = document.getElementById("search-result");
+        searchResult.innerHTML = `<p>Land bestaat niet</p>`;
         console.error(e);
     }
 }
 
-giveCountry();
+function countryCurrency (countryInfo) {
+    const currencyArray = countryInfo[0].currencies;
+    const currencyNames = [];
+    for (let i = 0; i < currencyArray.length; i++) {
+        currencyNames.push(currencyArray[i].name)
+    }
```

Comment:
Je zou dit iets kunnen versimpelen:
```suggestion
    const currencyNames = currencyArray.map(currency => currency.name);
```